src/libasr/codegen/asr_to_c.cpp

```diff
@@ -524,11 +529,7 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
             } else if (ASR::is_a<ASR::List_t>(*v_m_type)) {
                 ASR::List_t* t = ASR::down_cast<ASR::List_t>(v_m_type);
                 std::string list_type_c = c_ds_api->get_list_type(t);
-                std::string name = v.m_name;
-                if (v.m_intent == ASRUtils::intent_out) {
-                    name = "*" + name;
-                }
-                sub = format_type_c("", list_type_c, name,
+                sub = format_type_c("", list_type_c, v.m_name,
```

Comment:
This change should be reverted, we need those 4 lines as per this PR (https://github.com/lcompilers/lpython/commit/f9ab31ebc99f6a76f9d3d1e162e37e667e550d05)